pandas/core/aggregation.py

```diff
@@ -415,67 +423,94 @@ def transform(
     ValueError
         If the transform function fails or does not transform.
     """
-    from pandas.core.reshape.concat import concat
-
     is_series = obj.ndim == 1
-
     if obj._get_axis_number(axis) == 1:
         assert not is_series
         return transform(obj.T, func, 0, *args, **kwargs).T
 
-    if isinstance(func, list):
+    if is_list_like(func) and not is_dict_like(func):
```

Comment:
can you enhance the tests to hit is_list_like, e.g. a np.array

a Series is dict_like